Shard all of tests based on how many tests exist.

zdevito opened the pull request:
Summary: Reduces pytorch_core build from 2 hours to 30 minutes

Differential Revision: D10524261

ezyang commented on test/common_utils.py:
```diff
@@ -690,3 +691,109 @@ def random_fullrank_matrix_distinct_singular_value(l, *batches):
+    'test_Conv2d_no_bias',
+    'test_matmul_4d_4d',
+    'test_multinomial_invalid_probs',
+    'test_cuda_not_built',
```
Most of these exclusions make sense, but this one is pretty weird.

zdevito commented:
@pytorchbot retest this please

zdevito commented:
@pytorchbot retest this please

zdevito committed:
Shard all of tests based on how many tests exist. (#13160)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13160

Reduces pytorch_core build from 2 hours to 30 minutes

Reviewed By: soumith, dzhulgakov

Differential Revision: D10524261

fbshipit-source-id: 2bd62e428fdb156c70f36077f6b04afc13613119
